src/test/java/org/killbill/billing/plugin/adyen/api/TestAdyenPaymentPluginApi.java

```diff
@@ -414,6 +439,113 @@ public void testHPP() throws Exception {
+        final PaymentTransaction captureTransaction = TestUtils.buildPaymentTransaction(payment, TransactionType.CAPTURE, DEFAULT_CURRENCY);
+
+        final PaymentTransactionInfoPlugin authorizationInfoPlugin = adyenPaymentPluginApi.authorizePayment(payment.getAccountId(),
+                payment.getId(),
+                authorizationTransaction.getId(),
+                payment.getPaymentMethodId(),
+                authorizationTransaction.getAmount(),
+                authorizationTransaction.getCurrency(),
+                propertiesWith3DSInfo,
+                context);
+
+        assertEquals(authorizationInfoPlugin.getGatewayError(), "RedirectShopper");
+        final URL issuerUrl = new URL(PluginProperties.findPluginPropertyValue("issuerUrl", authorizationInfoPlugin.getProperties()));
+        final String md = PluginProperties.findPluginPropertyValue("MD", authorizationInfoPlugin.getProperties());
+        final String paReq = PluginProperties.findPluginPropertyValue("PaReq", authorizationInfoPlugin.getProperties());
+
+        final String responseHTML = given().log().all()
+                .contentType(ContentType.URLENC)
+                .accept(ContentType.HTML)
+                .formParam("MD", md)
+                .formParam("PaReq", paReq)
+                .formParam("TermUrl", DUMMY_URL)
+                .post(issuerUrl)
+                .then().log().all()
+                .statusCode(200)
+                .extract().asString();
+
+        final Map<String, String> formParams = extractForm(responseHTML);
+        assertFalse(formParams.isEmpty(), "No FORM found in HTML response");
+
+        final String formAction = rewriteFormURL(issuerUrl, formParams.remove("formAction"));
+        formParams.put("username", "user");
+        formParams.put("password", "password");
+
+        final String redirectHTML = given().log().all()
+                .contentType(ContentType.URLENC)
+                .accept(ContentType.HTML)
+                .formParams(formParams)
+                .post(formAction)
+                .then().log().all()
+                .statusCode(200)
+                .extract().asString();
+
+        final Map<String, String> redirectFormParams = extractForm(redirectHTML);
+        assertFalse(redirectFormParams.isEmpty(), "No FORM found in redirect HTML response");
+        assertEquals(DUMMY_URL, redirectFormParams.remove("formAction"));
+        redirectFormParams.put(PROPERTY_FROM_HPP, "true");
+
+        final List<PluginProperty> propertiesWithCompleteParams = PluginProperties.buildPluginProperties(redirectFormParams);
+
+        final Iterable<PluginProperty> mergedProperties = PluginProperties.merge(propertiesWith3DSInfo, propertiesWithCompleteParams);
+
+        final PaymentTransactionInfoPlugin purchaseInfoPlugin = adyenPaymentPluginApi.purchasePayment(payment.getAccountId(),
```

Comment:
@pierre triggering a second AUTH after the redirect comes back results in an Authorized response from Adyen, but then this Exception occurs:
```
org.jooq.exception.InvalidResultException: Cursor returned more than one result
	at org.jooq.impl.Utils.fetchOne(Utils.java:1093)
	at org.jooq.impl.AbstractResultQuery.fetchOne(AbstractResultQuery.java:440)
	at org.jooq.impl.SelectImpl.fetchOne(SelectImpl.java:2436)
	at org.killbill.billing.plugin.dao.payment.PluginPaymentDao$3.withConnection(PluginPaymentDao.java:155)
	at org.killbill.billing.plugin.dao.payment.PluginPaymentDao$3.withConnection(PluginPaymentDao.java:146)
	at org.killbill.billing.plugin.dao.PluginDao.execute(PluginDao.java:111)
	at org.killbill.billing.plugin.dao.payment.PluginPaymentDao.getSuccessfulAuthorizationResponse(PluginPaymentDao.java:145)
	at org.killbill.billing.plugin.adyen.api.AdyenPaymentPluginApi.executeFollowUpTransaction(AdyenPaymentPluginApi.java:410)
	at org.killbill.billing.plugin.adyen.api.AdyenPaymentPluginApi.capturePayment(AdyenPaymentPluginApi.java:209)
	at org.killbill.billing.plugin.adyen.api.TestAdyenPaymentPluginApi.testAuthorizeAndComplete3DSecure(TestAdyenPaymentPluginApi.java:510)
```